google/release/ha_image_janitor.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
+#
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import datetime
+import json
+import os
+import re
+import sys
+import yaml
+
+from google.cloud import storage
+from spinnaker.run import check_run_quick, run_quick
+
+
+RELEASED_VERSION_MATCHER = re.compile('^[0-9]+\.[0-9]+\.[0-9]+\.yml$')
+LATEST_UNVALIDATED_MATCHER = re.compile('.*-latest-unvalidated\.yml$')
+BUILT_VERSION_MATCHER = re.compile('.*-[0-9]+\-[0-9]+\-[0-9]+\-[0-9]+\.yml$')
+
+SERVICES = [
+  'clouddriver',
+  'deck',
+  'echo',
+  'front50',
+  'gate',
+  'igor',
+  'orca',
+  'rosco',
+  'fiat'
+]
+
+RELEASED_TAG_KEY = 'released'
+
+
+def __partition_boms(gcs_client, bucket_name):
+  bucket = gcs_client.get_bucket(bucket_name)
+  # Get all BOM blobs but remove 'bom/' directory.
+  all_bom_blobs = [b for b in bucket.list_blobs(prefix='bom') if not b.name.endswith('/')]
+  bom_contents_by_name = {__bom_to_version(b): b.download_as_string() for b in all_bom_blobs}
```

Comment:
This is a feature of the storage API -- the `name` of a blob is actually a path relative to the root bucket, which is why we need to use `basename` here. I changed the filter to look for `.yml` extensions.